magento/pwa-studio
# [PWA-1598] Add product to Cart from the wishlist

## Description

As a customer I want to add a wish-listed product to the cart *so that* I can buy a product that I saved for later earlier 
### Preconditions
 * Wishlist is enabled in Magento Admin Panel>Stores > Settings > Configuration > Customers > Wish List
 * Multi-wishlist is enabled in Stores > Settings > Configuration > Customers > Wish List>Enable Multiple Wish Lists (EE)
 * User is logged in to Venia and opens My Account>Favorites
 * User has products favorited

### Acceptance Criteria
 * User sees an option to add product to cart for each product in the wishlist. If configurable product is in the wishlist, user sees the option to configure it and add to the cart.
 * User selects adding product to the cart, cart gets updated and products stays in the favorite list. If user chooses to add the same product to the cart, product quantity gets updated.
 * If product in the favorites is out of stock, user sees an indication of that, no option to add it wishlist available.

### Test Plan
1. Add a Simple item to wishlist
2. Verify that you can add the simple item to the cart successfully without having to select any options
3. Verify that the item also remains in the wishlist
4. Add a configurable item with options selected to wishlist
5. Verify that you can add the configurable item to the cart successfully (have to confirm option selections?)
6. Verify that the item also remains in the wishlist
7. Add a configurable item with no options selected to wishlist
8. Verify that you can add the item to the cart successfully, having to first select item options via a dialog
9. Verify that the dialog works on mobile
10. Verify that the item remains in the wishlist after being added to cart
11. Mark one of the wishlist items as out of stock in the Magento Admin
12. Verify that the item's "Add to Cart" button is disabled. It should be visually different (grayed out), and not clickable

## Related Issue





* [[PWA-1598](https://jira.corp.magento.com/browse/PWA-1598)] Add product to Cart from the wishlist

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. See Test Plan

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
